src/EnergyPlus/SQLiteProcedures.cc

```diff
@@ -2060,442 +1992,768 @@ void SQLite::updateSQLiteSimulationRecord(
 
-		sqliteStepCommand(m_nominalElectricEquipmentInsertStmt);
-		sqliteResetCommand(m_nominalElectricEquipmentInsertStmt);
+		sqliteStepCommand(m_environmentPeriodInsertStmt);
+		sqliteResetCommand(m_environmentPeriodInsertStmt);
 	}
 }
 
-void SQLite::createSQLiteNominalGasEquipmentTable()
-{
-	for(int gasEquipNum = 1; gasEquipNum <= DataHeatBalance::TotGasEquip; ++gasEquipNum) {
-		auto const & gasEquipHB = DataHeatBalance::ZoneGas(gasEquipNum);
-		sqliteBindInteger(m_nominalGasEquipmentInsertStmt, 1, gasEquipNum);
-		sqliteBindText(m_nominalGasEquipmentInsertStmt, 2, gasEquipHB.Name);
-		sqliteBindForeignKey(m_nominalGasEquipmentInsertStmt, 3, gasEquipHB.ZonePtr);
-		sqliteBindForeignKey(m_nominalGasEquipmentInsertStmt, 4, gasEquipHB.SchedPtr);
-		sqliteBindDouble(m_nominalGasEquipmentInsertStmt, 5, gasEquipHB.DesignLevel);
-		sqliteBindDouble(m_nominalGasEquipmentInsertStmt, 6, gasEquipHB.FractionLatent);
-		sqliteBindDouble(m_nominalGasEquipmentInsertStmt, 7, gasEquipHB.FractionRadiant);
-		sqliteBindDouble(m_nominalGasEquipmentInsertStmt, 8, gasEquipHB.FractionLost);
-		sqliteBindDouble(m_nominalGasEquipmentInsertStmt, 9, gasEquipHB.FractionConvected);
-		sqliteBindText(m_nominalGasEquipmentInsertStmt, 10, gasEquipHB.EndUseSubcategory);
-
-		sqliteStepCommand(m_nominalGasEquipmentInsertStmt);
-		sqliteResetCommand(m_nominalGasEquipmentInsertStmt);
-	}
+void SQLite::addScheduleData( int const number, std::string const & name, std::string const & type, double const & minValue, double const & maxValue ) {
+	schedules.push_back(
+		std::unique_ptr<Schedule>(new Schedule(m_errorStream, m_db, number, name, type, minValue, maxValue))
+	);
 }
 
-void SQLite::createSQLiteNominalSteamEquipmentTable()
-{
-	for(int steamEquipNum = 1; steamEquipNum <= DataHeatBalance::TotStmEquip; ++steamEquipNum) {
-		auto const & steamEquipHB = DataHeatBalance::ZoneSteamEq(steamEquipNum);
-		sqliteBindInteger(m_nominalSteamEquipmentInsertStmt, 1, steamEquipNum);
-		sqliteBindText(m_nominalSteamEquipmentInsertStmt, 2, steamEquipHB.Name);
-		sqliteBindForeignKey(m_nominalSteamEquipmentInsertStmt, 3, steamEquipHB.ZonePtr);
-		sqliteBindForeignKey(m_nominalSteamEquipmentInsertStmt, 4, steamEquipHB.SchedPtr);
-		sqliteBindDouble(m_nominalSteamEquipmentInsertStmt, 5, steamEquipHB.DesignLevel);
-		sqliteBindDouble(m_nominalSteamEquipmentInsertStmt, 6, steamEquipHB.FractionLatent);
-		sqliteBindDouble(m_nominalSteamEquipmentInsertStmt, 7, steamEquipHB.FractionRadiant);
-		sqliteBindDouble(m_nominalSteamEquipmentInsertStmt, 8, steamEquipHB.FractionLost);
-		sqliteBindDouble(m_nominalSteamEquipmentInsertStmt, 9, steamEquipHB.FractionConvected);
-		sqliteBindText(m_nominalSteamEquipmentInsertStmt, 10, steamEquipHB.EndUseSubcategory);
+void SQLite::addZoneData( int const number, DataHeatBalance::ZoneData const & zoneData ) {
+	zones.push_back(
+		std::unique_ptr<Zone>(new Zone(m_errorStream, m_db, number, zoneData))
```

Comment:
The way you are doing this, one Zone instance per db record looks like Active Record pattern to me.  Is that what you are going for?  Why is the insertIntoSqlite a different step?  That is it seems you defer the actually committing to db.